src/Mod/Part/App/TopoShape.cpp

```diff
@@ -2503,6 +2503,62 @@ TopoDS_Shape TopoShape::makeLongHelix(Standard_Real pitch, Standard_Real height,
     return TopoDS_Shape(std::move(wire));
 }
 
+TopoDS_Shape TopoShape::makeSpiralHelix(Standard_Real radiusbottom, Standard_Real radiustop,
+                                  Standard_Real height, Standard_Real nbturns,
+                                  Standard_Real breakperiod, Standard_Boolean leftHanded) const
+{
+    // 1000 periods is an OCCT limit. The 3D curve gets truncated
+    // if he 2D curve spans beyond this limit.
+    if ((breakperiod < 0) || (breakperiod > 1000))
+        Standard_Failure::Raise("Break period must be in [0, 1000]");
+    if (breakperiod == 0)
+        breakperiod = 1000;
+    if (nbturns <= 0)
+        Standard_Failure::Raise("Number of turns must be greater than 0");
+
+    Standard_Real nbPeriods = nbturns/breakperiod;
+    Standard_Real nbFullPeriods = floor(nbPeriods);
+    Standard_Real partPeriod = nbPeriods - nbFullPeriods;
+
+    // A Bezier curve is used below, to get a periodic surface also for spirals.
+    TColgp_Array1OfPnt poles(1,2);
+    poles(1) = gp_Pnt(fabs(radiusbottom), 0, 0);
+    poles(2) = gp_Pnt(fabs(radiustop), 0, height);
+    Handle(Geom_BezierCurve) meridian;
+    meridian = new Geom_BezierCurve(poles);
+
+    gp_Ax1 axis(gp_Pnt(0.0,0.0,0.0) , gp::DZ());
+    Handle(Geom_Surface) surf;
+    surf = new Geom_SurfaceOfRevolution(meridian, axis);

```

Comment:
Where is this memory freed?

Reply:
??? What are talking about ? You're talking to a python dev ! ;D
... a suggestion please ?

Reply:
From what I see everywhere else in the file, there is no need to free this.
I think this is done by OCCT Handle

Reply:
I think you are right. The handle seems to be a kind of smart pointer. It was simply a doubt I had, and this answers my question thanks.